### PR TITLE
Add secrets-for-zowe-sdk dependency to ZE API

### PR DESCRIPTION
**Pull request by JillieBeanSim**
## Proposed changes

After some testing of importing ZE API as a dependency for extender since the @zowe/secrets-for-zowe-sdk was added and saw issues.  This PR adds the dependency to the package and moves the zowe-explorer's to top-level similar to how we handle the @zowe/cli dependency in the monorepo.  I also added the check in the ZE API build command script for the package.

## Release Notes

Milestone: 2.10.0

Changelog: n/a

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [x] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [x] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

## Further comments

Screen Shot 2023-08-18 at 12 48 57 PM

### CODE REVIEW

**Review by rudyflores — DISMISSED** [review]
Thanks for this change Billie!

**Review by zFernand0 — CHANGES_REQUESTED** [review]
LGTM! 😋 

TLDR: (if possible) I would like to see the CLI being exported from the ZE API
That's the reason for my requested changes.

We should merge this PR before Rudy's.
Definitely fixes the following issue in Rudy's PR: 😋 
```
[2023/08/21 13:16:28.483] [ERROR] [CredentialManagerFactory.js:151] Error: Failed 
to load Keytar module: Cannot find module '@zowe/secrets-for-zowe-sdk'
```

---

Also, I didn't know whether this is the right place to ask for this or not... but could we export an instance of `zowe` from the zowe-explorer-api?
That way, VSCode Extenders don't have to import from node_modules of the ZE API in order to bypass the annoying ProfileInfo error not being from the same location as the ZE API.

---

And there is one more thing to note (for Extenders)...
There are a few methods that could be broken given that the function signature of  `getProfileFromConfig` changed to include the `undefined`.

- How is that breaking?
  - It's possible for an extenders to have defined something of type `IProfAttrs` and TypeScript won't let them transpile/compile because it's not the same as the union of `IProfAttrs | undefined` 😋 

---

Last, but certainly not least...
In the event that an extenders has not updated their CLI dependency, we could be breaking them by introducing new optional method to the ZE API...
- How?
   - For instance the `dataSetsMatchingPattern` has a signature that includes an interface called `zowe.IDsmListOptions`, which was introduced in [`@zowe/cli@7.3.0`](https://github.com/zowe/zowe-cli/blob/master/packages/zosfiles/CHANGELOG.md#730)
     - the CICS extension still had 7.2.3 😋 

So, you can imagine if an extender has a version of the CLI lower than that, then they may get issues like this:
```
node_modules/@zowe/zowe-explorer-api/lib/profiles/ZoweExplorerApi.d.ts:297:67 - error 
TS2724: '"/root/gh/zowe/vscode-extension-for-cics/node_modules/@zowe/cli/lib/index"' 
has no exported member named 'IDsmListOptions'. Did you mean 'IUSSListOptions'?

297         dataSetsMatchingPattern?(filter: string[], options?: zowe.IDsmListOptions): 
Promise<zowe.IZosFilesResponse>;
```

I believe this could be prevented if the ZE API exported their version of `@zowe/cli` and extenders used that one, instead having another dependency in their package.json files

**Review by traeok — COMMENTED** [review]
Just left a quick comment on how the dependencies are structured, but everything LGTM. Thanks @JillieBeanSim

**Comment by t1m0thyj** [comment]
> Also, I didn't know whether this is the right place to ask for this or not... but could we export an instance of `zowe` from the zowe-explorer-api? That way, VSCode Extenders don't have to import from node_modules of the ZE API in order to bypass the annoying ProfileInfo error not being from the same location as the ZE API.

I like the idea of exporting the Imperative APIs from zowe-explorer-api to avoid errors related to `instanceof` 👍 Not sure if we want to export the entire CLI package from zowe-explorer-api, because that may prevent us from migrating to the SDK packages in Zowe v3?

Perhaps we could export just the `imperative` namespace from zowe-explorer-api, and also make the `@zowe/cli` package a peer dependency of zowe-explorer-api to address the other concerns about optional methods being added.

File: package.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
   },
   "dependencies": {
     "@zowe/cli": "7.18.0",
+    "@zowe/secrets-for-zowe-sdk": "7.18.1",
```
[the remark this answers is not in the document]
Comment (zFernand0) [reply package.json:19]: This resolved the runtime error caused by the `ProfilesCache.requireKeyring` from an extenders' perspective 😋

**Review by zFernand0 — APPROVED** [review]
This works perfectly fine for me using CICS.
I tested this PR in combination with Rudy's:
- #2397 

LGTM! 😋

**Review by traeok — APPROVED** [review]
LGTM, thanks for addressing this @JillieBeanSim!
